python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -59,9 +59,9 @@ def write(
     ) -> Any:
         import pyarrow.parquet as pq
 
-        blocks = list(blocks)
+        blocks = [b for b in blocks if BlockAccessor.for_block(b).num_rows() > 0]
 
-        if all(BlockAccessor.for_block(block).num_rows() == 0 for block in blocks):
+        if len(blocks) == 0:
```

Comment:
instead of filtering out empty blocks prior to `write_blocks_to_path()`, i think we should instead check for empty blocks in the [for loop here](https://github.com/ray-project/ray/pull/47270/files#diff-41d547362a05e79e9629e16f5ee08ba1fb4bc3872504e6529b422f86eba583d8L76-L78), and skip the `writer.write_table(table)` call if it is an empty block.